build.gradle

```diff
@@ -286,6 +286,14 @@ subprojects {
         }
     }
 
+    task testModules(type: Exec) {
+        dependsOn "jar"
+
+        commandLine "java", "-p", "${project.buildDir.absolutePath}${File.separator}libs${File.separator}/${project.name}-${project.version}.jar", "--list-modules"
```

Comment:
This: `libs${File.separator}/` will cause: `libs//` but I would not bother building the path like this, since the jar task has a `archivePath` property so you can do this:

```
task testModules(type: Exec) {
    dependsOn jar
    commandLine 'java', '-p', "$jar.archivePath", '--list-modules'
    standardOutput = new ByteArrayOutputStream()
}
```